### DISCUSSION TITLE
Add item random ID

### DISCUSSION THREAD

**Discussion by Russell500** [gp-seattle/rms-app#135] [Backend Questions] [answered]
![image](https://user-images.githubusercontent.com/87156867/128405254-7c1ed66d-af0f-4252-8f25-a0b1ef18c57a.png)
This is the code I have so far and I wanted to make sure that I'm pulling the right check to see if the item ID already exists if not I am not sure where to find the main storage of the IDs to be able to refer to it.

**Comment by yauj** [comment, marked as answer]
@Russell500 I would just copy the code that I linked to you. `this.itemTable.get` returns a promise, so then you need a .then to check that it actually returned a item. If it doesn't return a item, it would be null. I think it would look something like this:
```
private getUniqueId(): Promise<string> {
   const id = _Your Id Generation Logic_
   return this.itemTable.get()
       .then((item: ItemSchema) => {
           if (item) {
                 // Item Returned
                 return getUniqueId()
           } else {
                // No Item Returned
                return id;
           }
       })
}
```
